src/shared/unit-file.c

```diff
@@ -31,6 +31,41 @@ bool unit_type_may_template(UnitType type) {
                       UNIT_PATH);
 }
 
+int unit_dependency_name_compatible(const char *symlink, const char *target, bool instance_propagation) {
+        _cleanup_free_ char *template = NULL;
+        int r, un_type1, un_type2;
+
+        un_type1 = unit_name_classify(symlink);
+
+        /* The straightforward case: the symlink name matches the target and we have a valid unit */
+        if (streq(symlink, target) &&
+            un_type1 & (UNIT_NAME_PLAIN | UNIT_NAME_INSTANCE))
```

Comment:
given that this is not a plain boolean but a bit mask != 0 check, i'd prefer if we'd make that explicit, i.e. "(un_type & (…|…)) != 0"

Reply:
It's a "bitwise and", so still and "and". I added parentheses to make this clearer though.